pkg/gitrepo/repo.go

```diff
@@ -48,13 +51,14 @@ func New(config Config) (*Repo, error) {
 		}
 	}
 
-	fs := osfs.New(config.Dir)
-	storage := filesystem.NewStorageWithOptions(fs, cache.NewObjectLRUDefault(), filesystem.Options{ExclusiveAccess: true})
+	worktree := osfs.New(config.Dir)
+	fs := osfs.New(filepath.Join(config.Dir, ".git"))
+	storage := filesystem.NewStorageWithOptions(fs, cache.NewObjectLRUDefault(), filesystem.Options{})
```

Comment:
Why no longer exclusive access?

Reply:
Just in case. I thought it's premature optimization.